src/lib/utils/hooks/usePreFetchResource.ts

```diff
@@ -19,15 +18,16 @@ export function releaseResourceUrl(resourceUrl: string) {
 export default function usePreFetchResource(
```

Comment:
Split this up. Now, this hook takes a URL and returns a blob.

New fn `useCreateUrlForData` takes a Blob and returns a data URL.